lib/teleterm/apiserver/apiserver.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"net"
+	"strings"
+
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/teleport/lib/teleterm/apiserver/handler"
+
+	"github.com/gravitational/trace"
+
+	"google.golang.org/grpc"
+)
+
+// New creates an instance of API Server
+func New(cfg Config) (apiServer *APIServer, err error) {
+	if err := cfg.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	serviceHandler, err := handler.New(
+		handler.Config{
+			DaemonService: cfg.Daemon,
+		},
+	)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	ls, err := newListener(cfg.HostAddr)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	grpcServer := grpc.NewServer(grpc.Creds(nil), grpc.ChainUnaryInterceptor(
+		withErrorHandling(cfg.Log),
+	))
+
+	api.RegisterTerminalServiceServer(grpcServer, serviceHandler)
+
+	return &APIServer{cfg, ls, grpcServer}, nil
+}
+
+// Serve starts accepting incoming connections
+func (s *APIServer) Serve() error {
+	return s.grpcServer.Serve(s.ls)
+}
+
+// Stop stops the server and closes all listeners
+func (s *APIServer) Stop() {
+	s.grpcServer.GracefulStop()
+}
+
+func newListener(host string) (net.Listener, error) {
+	var network, addr string
+
+	parts := strings.SplitN(host, "://", 2)
```

Comment:
`://` indicates a URL to me, so I would consider renaming the `host` argument.

Also, would it be best to use `url.Parse` rather than raw string parsing?

Reply:
Addressed in 634d99f95c837d8.